### DISCUSSION TITLE
Teleport when above nether ceiling

### DISCUSSION THREAD

**Discussion by 1ByteBit** [PurpurMC/Purpur#1187] [__Completed__] [answered]
An option to teleport a player if they go above the nether ceiling, and I believe there's a few ways to do this:
- If a player uses a chorus fruit or enderpearl to go through the bedrock, teleport them back to their original position
- If a player goes above the ceiling, teleport them to the closest safe area
- Just straight up teleport them directly below the bedrock layers, you can do a couple things with this such as removing the netherrack/blocks they spawn inside if needed, and placing a netherrack block directly below them if needed. Neither of those "safety checks" are required however

**Comment by Roan-V** [comment]
I believe this setting will do that https://purpurmc.org/docs/Configuration/teleport-if-outside-border
Alternatively https://docs.papermc.io/paper/reference/world-configuration#nether-ceiling-void-damage-height

**Comment by BillyGalbreath** [comment, marked as answer]
Implemented an option to teleport the player to spawn (nether spawn, not server spawn) when they take nether ceiling damage in build 1841.

https://github.com/PurpurMC/Purpur/commit/b3d374f819851f79d1e8eebb496ec66a9bcd2c68
